compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1836,7 +1838,7 @@ class Definitions {
     tp.stripTypeVar.dealias match
       case tp1: TypeParamRef if ctx.typerState.constraint.contains(tp1) =>
         asContextFunctionType(TypeComparer.bounds(tp1).hiBound)
-      case tp1 @ RefinedType(parent, nme.apply, mt: MethodType) if isErasedFunctionType(parent) && mt.isContextualMethod =>
+      case tp1 @ PolyOrErasedFunctionOf(mt: MethodType) if mt.isContextualMethod =>
```

Comment:
I see that this only matches ErasedFunction and not PolyFunction because it takes a MethodType and not a PolyType, but this could easily lead to confusion (especially if we have polymorphic ErasedFunctions in the future), so I would prefer to have explicit ErasedFunctionOf/PolyFunctionOf extractors.

Reply:
Added `ErasedFunctionOf`. There are no places left where we would use `PolyFunctionOf`.